10_rest/mohitnag/pkg/rest/rest.go

```diff
@@ -0,0 +1,121 @@
+package rest
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	"github.com/anz-bank/go-course/10_rest/mohitnag/pkg/puppy"
+	"github.com/go-chi/chi"
+	"github.com/go-chi/chi/middleware"
+	"github.com/go-chi/render"
+)
+
+type Handler struct {
+	storer puppy.Storer
+	router chi.Router
+}
+
+func (rh *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	rh.router.ServeHTTP(w, r)
+}
+
+func NewRestHandler(s puppy.Storer) http.Handler {
+	r := chi.NewRouter()
+	r.Use(middleware.RequestID)
+	r.Use(middleware.Logger)
+	r.Use(middleware.Recoverer)
+	r.Use(middleware.URLFormat)
+	r.Use(render.SetContentType(render.ContentTypeJSON))
+	rh := Handler{
+		storer: s,
+		router: r}
+	rh.handleRequests()
+	return &rh
+}
+
+func (rh *Handler) handleRequests() {
+	r := rh.router
+	r.Route("/api/puppy", func(r chi.Router) {
+		r.Route("/", func(r chi.Router) {
+			r.Post("/", rh.handlePostPuppy)
+		})
+		r.Route("/{id}", func(r chi.Router) {
+			r.Get("/", rh.handleGetPuppy)
+			r.Put("/", rh.handleUpdatePuppy)
+			r.Delete("/", rh.handleDeletePuppy)
+		})
+	})
+}
+
+func (rh *Handler) handlePostPuppy(w http.ResponseWriter, r *http.Request) {
+	var puppy puppy.Puppy
+	if err := render.DecodeJSON(r.Body, &puppy); err != nil {
+		renderDecodeJSONErrorResponse(w, r, err)
+		return
+	}
+	if err := rh.storer.CreatePuppy(puppy); err != nil {
+		renderErrorResponse(w, r, err)
+		return
+	}
+	w.WriteHeader(http.StatusCreated)
+}
+
+func (rh *Handler) handleGetPuppy(w http.ResponseWriter, r *http.Request) {
+	var puppy puppy.Puppy
+	ID := chi.URLParam(r, "id")
+	puppyID, _ := strconv.Atoi(ID)
```

Comment:
You need to handle errors from `strconv.Atoi`, because this is unsafe user input. Do not ignore errors that come from parsing user input - that leads to security issues.

Reply:
Done. Please have a look.